lib/sort_by_length.rb

```diff
@@ -1,7 +1,31 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n3), cubic. nested loops, each of them are O(n)
+# Space complexity: O(n), creating new data structure when splitting
 def sort_by_length(my_sentence)
-  raise NotImplementedError, "Method not implemented"
+  #raise NotImplementedError, "Method not implemented"
+
+  return '' if my_sentence.nil?
+
+  split_arr = my_sentence.split(' ')
+  length = split_arr.length
+  i = 0
+  while i < length - 1 #outer
+    i = 0
+    while i < length - 1 # inner loop
```

Comment:
Look at this inner and outer loop, they both depend and change `i`.  So these really function as 1 loop.  You only need 2 loops total!